dist/s_evergreen

```diff
@@ -16,6 +16,31 @@ test -s $t && {
     cat $t
     echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
 
+    echo "Check for Python dependencies without specified versions:"
+
+    unspecified_dependency=false
+
+    pip_installs=$(grep -E 'pip(3)? install' "test/evergreen.yml")
+
+    while IFS= read -r line; do
+        for word in $line; do
+            if [[ "$word" =~ ^pip(3)?$ || "$word" == "install" || "$word" =~ ^python(3)?$ || "$word" =~ -.* ]]; then
+                continue
+            fi
+            if [[ ! "$word" =~ .*==.* ]]; then
+                unspecified_dependency=true
+                echo "Version not specified for Python dependency $word in line: $line"
+                exit 1
```

Comment:
If we exit here we'll only catch the first occurrence. I'd suggest we remove this exit and have a 
```
if  unspecified_dependency; then
    exit 1
fi
```
check below